src/unix/fs.c

```diff
@@ -1262,7 +1263,19 @@ static ssize_t uv__fs_copyfile(uv_fs_t* req) {
   if (req->flags & UV_FS_COPYFILE_EXCL)
     flags |= COPYFILE_EXCL;
 
-  if (copyfile(req->path, req->new_path, NULL, flags))
+  src = (char *)req->path;
+  if (flags & (COPYFILE_CLONE | COPYFILE_CLONE_FORCE)) {
+    /* COPYFILE_CLONE_* implies COPYFILE_NOFOLLOW_SRC, so manually resolve
+       src if it is a link. */
+    if (!uv__fs_readlink(req)) {
```

Comment:
I don't think this is an appropriate fix as it introduces a race condition (file can change between readlink and copyfile.) Hopefully there's another solution?

Reply:
The only solution I can think of is to revert the change that added the call to `copyfile()`.

If this function needs to be atomic then we either use `copyfile()` or not at all.

Reply:
Also doesn't the Linux branch have similar race conditions?

Reply:
I don't think so but, even it does, that's still no reason to introduce them here.

Is it an idea to fall back to the read/write loop (the one that's used on linux and other platforms) when one of those flags is set?

Reply:
I can give that a try. Macos also has `clonefile()` that I can look into.

Reply:
@lewis6991 here is a prior proposed PR using `clonefile()` - https://github.com/libuv/libuv/pull/3091

Reply:
> I don't think `copyfile()` maps properly to this API.

Welp, there's a strong irony when the thing's named after (and indeed originally inspired by, in issue 925) `copyfile()`.

Reply:
Ok, so it seems `copyfile()` is just unfinished. Seems like the best option for now is defaulting to the general implementation when any clone flags are present/

Reply:
Alternative would be to use `clonefile` _without_ the `CLONE_NOFOLLOW` flag set?

Reply:
What about force?

Reply:
I think `clonefile` behaves like force? (Not 100% sure, though.)

Reply:
I don't know these APIs's well enough. For now, I've just detected the regression case and defaulted to the unix impl.

Reply:
@clason is correct. the linked bit of code has its caller check for the error to bail out in the force case. https://github.com/apple-oss-distributions/copyfile/blob/ef8ca39c336c4c587b98ac8b33362409a122cb3e/copyfile.c#L1224